core/src/main/scala/scalaz/TheseT.scala

```diff
@@ -0,0 +1,148 @@
+  def onlyBoth(implicit F: Functor[F]): OptionT[F, (A, B)] = OptionT(F.map(run)(_.onlyBoth))
+  def pad(implicit F: Functor[F]): F[(Option[A], Option[B])] = F.map(run)(_.pad)
+  def fold[X](s: A => X, t: B => X, q: (A, B) => X)(implicit F: Functor[F]): F[X] = F.map(run)(_.fold(s, t, q))
+
+  def exists(p: B => Boolean)(implicit F: Functor[F]): F[Boolean] = F.map(run)(_ exists p)
+  def forall(p: B => Boolean)(implicit F: Functor[F]): F[Boolean] = F.map(run)(_ forall p)
+  def toList(implicit F: Functor[F]): ListT[F, B] = ListT(F.map(run)(_.toList))
+
+  def getOrElse(default: => B)(implicit F: Functor[F]): F[B]
+  = this.b.getOrElse(default)
+  def |(default: => B)(implicit F: Functor[F]): F[B]
+  = getOrElse(default)
+
+  def valueOr[BB >: B](x: A => BB)(implicit M: Semigroup[BB], F: Functor[F]): F[BB]
+  = F.map(run)(_ valueOr x)
+
+  def swapped[AA, BB](k: (B \&/ A) => (BB \&/ AA))(implicit F: Functor[F]): TheseT[F, AA,  BB] =
+    TheseT(F.map(run)(_.swapped(k)))
+
+  def ~[AA, BB](k: (B \&/ A) => (BB \&/ AA))(implicit F: Functor[F]): TheseT[F, AA,  BB] =
+    swapped(k)
+
+  def a(implicit F: Functor[F]): OptionT[F, A] = OptionT(F.map(run)(_.a))
+  def b(implicit F: Functor[F]): OptionT[F, B] = OptionT(F.map(run)(_.b))
+
+  def append[AA >: A, BB >: B](that: => TheseT[F, AA,  BB])(implicit F: Apply[F], SA: Semigroup[AA], SB: Semigroup[BB]): TheseT[F, AA,  BB]
+  = TheseT(F.apply2(this.run, that.run)(_ append _))
+
+  def leftMap[C](f: A => C)(implicit F: Functor[F]): TheseT[F, C, B]
+  = TheseT(F.map(run)(_.leftMap(f)))
+
+  def bimap[C, D](f: A => C, g: B => D)(implicit F: Functor[F]): TheseT[F, C, D]
+  = TheseT(F.map(run)(_.bimap(f, g)))
+
+  def traverse[G[_], AA >: A, D](g: B => G[D])(implicit F: Traverse[F], G: Applicative[G]): G[TheseT[F, AA, D]]
+  = G.map(F.traverse(run)(o => Traverse[AA \&/ ?].traverse(o)(g)))(TheseT(_))
+
+
+  def bitraverse[G[_], C, D](f: A => G[C], g: B => G[D])(implicit F: Traverse[F], G: Applicative[G]): G[TheseT[F, C, D]]
+  = G.map(F.traverse(run)(Bitraverse[\&/].bitraverseF(f, g)))(TheseT(_: F[C \&/ D]))
+
+  def &&&[AA >: A, C](t: TheseT[F, AA, C])(implicit M: Semigroup[AA], F: Apply[F]): TheseT[F, AA, (B, C)]
+  = TheseT(F.apply2(run, t.run)(_ &&& _))
+
+  def ===[AA >: A, BB >: B](x: TheseT[F, AA, BB])(implicit EA: Equal[AA], EB: Equal[BB], F: Apply[F]): F[Boolean]
+  = F.apply2(this.run, x.run)(_ === _)
+
+  def show(implicit SA: Show[A], SB: Show[B], F: Functor[F]): F[Cord]
+  = F.map(run)(_.show)
+
+  def foldRight[Z](z: => Z)(f: (B, => Z) => Z)(implicit F: Foldable[F]): Z
+  = F.foldRight[A \&/ B, Z](run, z)((a, b) => a.foldRight(b)(f))
+
+
+}
+
+
+sealed abstract class TheseTInstances1 {
+  implicit def TheseTInstance1[F[_]: Traverse, L]: Traverse[TheseT[F, L, ?]]
+  = new Traverse[TheseT[F, L, ?]] {
+    override def traverseImpl[G[_], A, B](fa: TheseT[F, L, A])(f: (A) => G[B])(implicit A: Applicative[G]) =
+      fa.traverse(f)
+
+  }
+
+  /* cannot use kind projector */
```

Comment:
That doesn't compile

    info] Compiling 1 Scala source to C:\Users\cmarsha\Work\temp\scalaz\core\target\scala-2.10\classes...
    error] C:\Users\cmarsha\Work\temp\scalaz\core\src\main\scala\scalaz\TheseT.scala:101: not found: type ?
    error]   implicit def TheseTHoist[A: Semigroup] = new Hoist[TheseT[?[_], A, ?]] {
    error]                                                             ^
    error] [X_kp2]scalaz.TheseT[<error>,A,X_kp2] takes one type parameter, expected: two
    error] [X_kp2]scalaz.TheseT[<error>,A,X_kp2] takes one type parameter, expected: two
    error] three errors found
    error] (core/compile:compile) Compilation failed